### DISCUSSION TITLE
Interpolate dashboard variable when used in annotation

### DISCUSSION THREAD

**Discussion by Mario-Hofstaetter** [grafana/grafana#38252] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

When adding a new annotation, it would be nice if entered **variable names (`$job` etc.) would be interpolated before saving**.
**Currently text and tags of an annotation will literally save the string `$job`.
Using syntax `${job}` or `[[job]]` also does not work.**

![grafik](https://user-images.githubusercontent.com/33002073/113462151-d7c76c80-941f-11eb-8135-e5969cbab10b.png)

After saving:
![grafik](https://user-images.githubusercontent.com/33002073/113462155-dd24b700-941f-11eb-9162-7fbedae92895.png)

If the feature is added, it should also work with [global variables](https://grafana.com/docs/grafana/latest/variables/variable-types/global-variables) like `$__dashboard` and others where applicable (e.g. `$__rate_interval` if chart uses Prometheus datasource)

**Edit 2021-05-11:**

It would be also super nice, if the annotation text would support Auto-Completion for variables (like the query editor sometimes, e.g. prometheus range variables), as soon as the user types `$` for example.

**Why is this needed**:

Many dashboards use variables like `$job`, `$instance` and so on.
When displaying annotations, one may wants to filter to only display annotations (alert) for the selected job (otherwise noisy).
For filtering it is necessary to add **Tags** to annotations.
Or the user just wants to add a Tag with a certain value he selected in a dashboard variable.

Variable values may be long (`really-long-server-hostname.company.com:9100`) which makes manual input hard or impossible.

**Desired behavior**:

Doing this:
![vH5DfQBeRB](https://user-images.githubusercontent.com/33002073/113462400-f37f4280-9420-11eb-8df8-c85f6897c2b5.gif)

Should result in this:
![grafik](https://user-images.githubusercontent.com/33002073/113462405-fbd77d80-9420-11eb-8ef5-b72bca0f1730.png)

**Playground**:

- Use any dashboard, e.g. <https://play.grafana.org/d/000000010/annotations>
- Get a variable:
![grafik](https://user-images.githubusercontent.com/33002073/113462458-4b1dae00-9421-11eb-97d9-5bb4ca5585ed.png)
- Add annotations and try using the variable

**Related**:

Related, but I don't want _automatically_ added tags: #24674

**Comment by NerijusRazvodovskis** [comment]
Any updates on this one?

**Comment by Mario-Hofstaetter** [comment]
Added nice-to-have feature in OP for variable auto-completion while entering in annotation text editor.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by siegenthalerroger** [comment]
This appears to actually work now, just tried it :)

Just used a `tag:$value` with a variable that was a text input. Just requires a data refresh before showing the alerts obviously

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
